File: rb/features.rb
# ParlamentOpenData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ParlamentOpenDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      ParlamentOpenDataBaseFeature.new
    when "test"
      ParlamentOpenDataTestFeature.new
    else
      ParlamentOpenDataBaseFeature.new
    end
  end
end

File: py/parlamentopendata_sdk/features.py
# ParlamentOpenData SDK feature factory

from parlamentopendata_sdk.feature.base_feature import ParlamentOpenDataBaseFeature
from parlamentopendata_sdk.feature.test_feature import ParlamentOpenDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ParlamentOpenDataBaseFeature(),
        "test": lambda: ParlamentOpenDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
